[xCOLOURx] iP

xCOLOURx opened the pull request:
## NOAC
>“You can always change you plan, but only if you have one.” - Randy Pausch, the Last Lecture [(source)](https://www.goodreads.com/quotes/124282-you-can-always-change-you-plan-but-only-if-you)

NOAC keeps track of all your plans so you never have to worry. Its,
- Text-based
- Easy to use
- Always reliable
- ~~FAST~~ ~~SUPER FAST~~ EXTREMELY FAST to use

All you need to do is,

1. Download it from [here](https://github.com/xCOLOURx/ip)
2. Run it
3. Add your task
4. Noac will settle the rest for you 😜 

And best of all it is **FREE**🔥 

Features:

- [x] Add tasks
- [ ] Edit tasks (coming soon)
- [ ] Reminders (coming soon)


If your are a Java programmer, you can use it to  practice Java too. You can find the `main` method under `Noac.java`:

```java
public static void main(String[] args) {
    new Noac("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

xCOLOURx committed:
Initial commit for lvl 0

xCOLOURx committed:
Initial commit for lvl 1

xCOLOURx committed:
Inital commit for lvl 2

xCOLOURx committed:
Inital commit for lvl 3

xCOLOURx committed:
Initial commit for lvl 4

xCOLOURx committed:
Initial commit for Automated Text UI Testing

xCOLOURx committed:
Initial commit for lvl 5

xCOLOURx committed:
Initial commit for lvl 6

xCOLOURx committed:
Add Save functionality

xCOLOURx committed:
Add Date and Time functionality

xCOLOURx committed:
Merge branch 'branch-level-8'

xCOLOURx committed:
Add more OOP

xCOLOURx committed:
Add Packages

xCOLOURx committed:
Merge branch 'add-gradle-support'

xCOLOURx committed:
Add Gradle Support

xCOLOURx committed:
Add JUnit Test

xCOLOURx committed:
Add JavaDocs comments

xCOLOURx committed:
Merge branch 'branch-A-JavaDoc'

xCOLOURx committed:
Fix lines according to coding standards

xCOLOURx committed:
Merge branch-A-CodingStandard into master

xCOLOURx committed:
Add find command

xCOLOURx committed:
Fix merge conflicts between master and branch-Level-9

emzm2023 commented on src/main/java/noac/Parser.java:
```diff
@@ -0,0 +1,218 @@
+        case "unmark":
+
+            if (userInputArr.length > 2 || userInputArr.length <= 1) {
+                throw new NoacException("☹ OOPS!!! Please enter in the format mark [TASK_NUMBER] e.g. mark 1");
```
I like your formatting of the switch statement!

emzm2023 commented on src/main/java/noac/Storage.java:
```diff
@@ -0,0 +1,144 @@
+     *
+     * @return The loaded tasks.
+     * @throws NoacException For any errors that needs to be displayed to user.
+     */
```
I like how clear and understandable your comments are

emzm2023 commented on src/main/java/noac/Ui.java:
```diff
@@ -0,0 +1,179 @@
+     *
+     * @param tasks The task to be displayed.
+     */
+    public void showOnDate(ArrayList<Task> tasks) {
```
Perhaps you could rename the method to be more self explanatory? eg. showTasksOnDate

xCOLOURx replied:
Thank you for your suggestion, I think it improves the clarity of my code

emzm2023 commented on src/main/java/noac/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+import noac.TaskList;
+import noac.Ui;
+
+/**
+ * Abstract class for all commands to extend from which includes the abstract
+ * execute function and also the isExit function.
+ */
```
I really like this description of the abstract class as it is clear, succinct, and detailed!

emzm2023 commented on src/main/java/noac/command/ExitCommand.java:
```diff
@@ -0,0 +1,37 @@
+ */
+public class ExitCommand extends Command {
+
+    private boolean isExit = true;
```
I like the naming of this boolean

emzm2023 reviewed (commented):
Your code is well commented and easily understandable!

chewjh1234 commented on src/main/java/noac/Parser.java:
```diff
@@ -0,0 +1,218 @@
+            String description = "";
+            String by = "";
+
+            boolean afterBy = false;
```
I am not sure what afterBy as a boolean means. Perhaps this boolean variable can have a name starting with is/was/has

chewjh1234 commented on src/main/java/noac/Parser.java:
```diff
@@ -0,0 +1,218 @@
+    public static LocalDateTime parseDate(String date) throws DateTimeParseException {
+        LocalDateTime localDateTime;
+
+        if(date.length() == 10) {
```
There should be a whitespace after `if`

chewjh1234 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,36 @@
-| | | | | | | |/ / _ \
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+ _   _  ___    _    ____
+| \ | |/ _ \  / \  / ___|
+|  \| | | | |/ _ \| |
+| |\  | |_| / ___ \ |___
+|_| \_|\___/_/   \_\____|
```
Great effort in coming up with this! Absolutely love it!

chewjh1234 commented on src/test/java/noac/task/DeadlineTest.java:
```diff
@@ -0,0 +1,42 @@
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+
```
Perhaps you can cut down on the number of blank lines here and in the rest of your code. One blank line has improved your code readability, but I doubt that 3 blank lines may do better.

chewjh1234 commented on src/main/java/noac/task/Event.java:
```diff
@@ -0,0 +1,68 @@
+        this.to = to;
+    }
+
+    /**
+     * Converts the task to string.
+     *
+     * @return The string.
+     */
+    @Override
```
Generally, JavaDocs aren't required for overriden methods. Great effort though!

chewjh1234 commented on src/main/java/noac/Storage.java:
```diff
@@ -0,0 +1,144 @@
+            FileWriter fileWriter = new FileWriter(filePath);
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+            for(int i = 0; i < taskList.size(); i++) {
```
There should be a whitespace after `for`

chewjh1234 commented on src/main/java/noac/Storage.java:
```diff
@@ -0,0 +1,144 @@
+                        }
+
+                        Event event;
+                        try{
```
There should be a whitespace after `try`

chewjh1234 commented on src/main/java/noac/Storage.java:
```diff
@@ -0,0 +1,144 @@
+import java.time.format.DateTimeParseException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
```
I like how your imports are categorised and then grouped together in alphabetical order. It makes it so much easier for everyone to read 😄

chewjh1234 reviewed (commented):
Your code looks good to me! And I really appreciate the effort you have put into creating the chatbot logo. You might need to fix some whitespaces here and there, as well as the naming of your variables, which I believe are minor fixes 😄

xCOLOURx committed:
add CheckStyle

xCOLOURx committed:
Add JavaFX GUI

xCOLOURx committed:
Add Assertions

Assertions not implemented.

Assertions helps to identify bugs in the code.

Assertions statements are added to the Parse.java file.

Parsing of the input contains the most assumptions and the assert statements can help identify the bugs.

xCOLOURx committed:
Merge pull request #2 from xCOLOURx/branch-A-Assertions

Add Assertions

xCOLOURx committed:
Refactor according to Coding Standards

Current Code is not up to Coding Standards

Refactor the code by implementing SLAP, avoiding long methods, magic
numbers and making happy path prominent, mainly in the Parser.java and
Storage.java files.

This makes the code alot more readable to other users.

xCOLOURx committed:
Merge pull request #3 from xCOLOURx/branch-A-CodeQuality

Refactor according to Coding Standards

xCOLOURx committed:
Setup CI

xCOLOURx committed:
Add Tagging

xCOLOURx committed:
Fix storage bug and Cleanup code

xCOLOURx committed:
Update main.yml

xCOLOURx committed:
Add User Guide and Screenshot

xCOLOURx committed:
Revert "Add User Guide and Screenshot"

This reverts commit 6cf0a4596ff09ed0f30023e195b1116ab946af4b.

xCOLOURx committed:
Merge branch 'master' of https://github.com/xCOLOURx/ip

xCOLOURx committed:
Add User Guide and Screenshot

xCOLOURx committed:
Fix delete bug
